lib/iris/fileformats/netcdf/saver.py

```diff
@@ -2773,26 +2801,117 @@ def save(
+                    # Replace the input cube with a copy, so we can modify attributes.
+                    cube = cube.copy()
+                    cubes[i_cube] = cube
+                    # Catch any demoted attrs where there is already a local version
+                    blocked_attrs = [
+                        attrname
+                        for attrname in demote_attrs
+                        if attrname in cube.attributes.locals
+                    ]
+                    if blocked_attrs:
+                        warnings.warn(
+                            f"Global cube attributes {blocked_attrs} "
+                            f'of cube "{cube.name()}" have been lost, overlaid '
+                            "by existing local attributes with the same names."
+                        )
+                    for attr in demote_attrs:
+                        if attr not in blocked_attrs:
+                            cube.attributes.locals[
+                                attr
+                            ] = cube.attributes.globals[attr]
+                        cube.attributes.globals.pop(attr)
+
     else:
-        local_keys = set(local_keys)
-
-    # Determine the attribute keys that are common across all cubes and
-    # thereby extend the collection of local_keys for attributes
-    # that should be attributes on data variables.
-    attributes = cubes[0].attributes
-    common_keys = set(attributes)
-    for cube in cubes[1:]:
-        keys = set(cube.attributes)
-        local_keys.update(keys.symmetric_difference(common_keys))
-        common_keys.intersection_update(keys)
-        different_value_keys = []
-        for key in common_keys:
-            if np.any(attributes[key] != cube.attributes[key]):
-                different_value_keys.append(key)
-        common_keys.difference_update(different_value_keys)
-        local_keys.update(different_value_keys)
+        # Determine the attribute keys that are common across all cubes and
+        # thereby extend the collection of local_keys for attributes
+        # that should be attributes on data variables.
+        # NOTE: in 'legacy' mode, this code derives a common value for 'local_keys', which
+        # is employed in saving each cube.
+        # However, in `split_attrs` mode, this considers ONLY global attributes, and the
+        # resulting 'common_keys' is the fixed result : each cube is then saved like ...
+        # "sman.write(... localkeys=list(cube.attributes) - common_keys, ...)"
+        if local_keys is None:
+            local_keys = set()
+        else:
+            local_keys = set(local_keys)
+
+        common_attr_values = None
+        for cube in cubes:
+            cube_attributes = cube.attributes
+            keys = set(cube_attributes)
+            if common_attr_values is None:
+                common_attr_values = cube_attributes.copy()
+                common_keys = keys.copy()
```

Comment:
I think it's only True the first time around the loop (i.e. when dealing with the first input cube).
But since I decided _not_ to modify this code, but instead re-implement for the save-split-attrs case, I can revert this code block  to [what it was before](https://github.com/SciTools/iris/blob/v3.6.1/lib/iris/fileformats/netcdf/saver.py#L2704-L2711) ...
I'll fix that

Reply:
OK I see I was wrong about point number 1.

But has something weird happened while restoring the code? It looks like it has been duplicated?

Reply:
Yes something weird.  Looking into it ...